hls-plugin-api/src/Ide/PluginUtils.hs

```diff
@@ -255,13 +256,18 @@ getNormalizedFilePath (PluginId plId) docId = handleMaybe errMsg
         uri' = docId ^. uri
 
 -- ---------------------------------------------------------------------
+throwPluginError :: Monad m => PluginId -> String -> String -> ExceptT String m b
+throwPluginError (PluginId who) what where' = throwE msg
+    where
+        msg = (T.unpack who) <> " failed with " <> what <> " at " <> where'
+
 handleMaybe :: Monad m => e -> Maybe b -> ExceptT e m b
 handleMaybe msg = maybe (throwE msg) return
 
 handleMaybeM :: Monad m => e -> m (Maybe b) -> ExceptT e m b
 handleMaybeM msg act = maybeM (throwE msg) return $ lift act
 
-response :: Monad m => ExceptT String m a -> m (Either ResponseError a)
-response =
+pluginResponse :: Monad m => ExceptT String m a -> m (Either ResponseError a)
+pluginResponse =
```

Comment:
Is there any gain to swap `ExceptT String m a` to `ExceptT Text m a`? I briefly experimented with it and it ripples up throughout most plugins, but not sure if it's even worth it?

Reply:
Figured as much. Maybe I'll look into it down the line.